File: app/src/main/java/com/fallgamlet/dnestrcinema/utils/LogUtils.java
package com.fallgamlet.dnestrcinema.utils;

import android.util.Log;

import com.fallgamlet.dnestrcinema.BuildConfig;

/**
 * Created by fallgamlet on 09.04.17.
 */

public class LogUtils {
    public static final boolean DEBUG = BuildConfig.DEBUG;

    public static void log(String msg) {
        log("", msg);
    }

    public static void log(String tag, String msg) {
        if (DEBUG) {
            Log.d(tag, msg);
        }
    }

    public static void log(String tag, String msg, Throwable throwable) {
        if (DEBUG) {
            Log.d(tag, msg, throwable);
        }
    }
}

File: app/src/main/java/com/fallgamlet/dnestrcinema/utils/LogUtils.kt
package com.fallgamlet.dnestrcinema.utils

import android.util.Log

import com.fallgamlet.dnestrcinema.BuildConfig
import com.google.gson.Gson
import com.google.gson.GsonBuilder


object LogUtils {
    val isDebug = BuildConfig.DEBUG

    private val gson = GsonBuilder()
            .setDateFormat("yyyy-MM-dd'T'HH:mm:ss.s'Z'")
            .create()

    fun log(tag: String, obj: Any?) {
        if(!isDebug) return

        log(tag, gson.toJson(obj))
    }

    fun log(msg: String) {
        log("", msg)
    }

    fun log(tag: String, msg: String?) {
        if(!isDebug) return

        Log.d(tag, msg?: "(null)")
    }

    fun log(tag: String, msg: String?, throwable: Throwable) {
        if(!isDebug) return

        Log.d(tag, msg?: "(null)", throwable)
    }
}
